regression/cbmc/union/union_member.c

```diff
@@ -0,0 +1,12 @@
+#include <assert.h>
+
+struct s {
+    union U {
+      char b[2];
+    };
+};
+
+int main() {
+  struct s X;
+  __CPROVER_assert(X.b[1] == 2, "should fail");
+}
```

Comment:
Won't fail anymore with #7230, but even now only fails with z3. Works fine with the in-tree SMT solver.